providers/azure.go

```diff
@@ -74,6 +75,9 @@ func NewAzureProvider(p *ProviderData) *AzureProvider {
 	if p.ProtectedResource == nil || p.ProtectedResource.String() == "" {
 		p.ProtectedResource = azureDefaultProtectResourceURL
 	}
+	if p.ValidateURL == nil || p.ValidateURL.String() == "" {
```

Comment:
Can we just set this a few lines above in `p.setProviderDefaults(providerDefaults{...` where the `validateURL` is currently set to `nil`?

Reply:
Done

Reply:
I had to remove this change again as it made `TestNewAzureProvider` fail now. Looking at the code, I realized that we can't simply rely on `p.setProviderDefaults` here, because the default value is dependent on `p.ProfileURL` being defaulted or not.